connectors/memory/memory.go

```diff
@@ -326,13 +334,51 @@ func (c *Connector) Remove(_ context.Context, ei *dosa.EntityInfo, values map[st
 	return nil
 }
 
+// RemoveRange removes all of the elements in the range specified by the entity info and the column conditions.
+func (c *Connector) RemoveRange(_ context.Context, ei *dosa.EntityInfo, columnConditions map[string][]*dosa.Condition) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	partitionRange := c.findRange(ei, columnConditions)
+	if partitionRange == nil {
+		return nil
+	}
+
+	partitionRef := partitionRange.entityRef[partitionRange.partitionKey]
```

Comment:
If you don't like the pointer approach, you can return the updated value with the range deleted (I think the pointer approach is more elegant though)